### PR TITLE
Add JUnit Testing For Components

### LINKED ISSUE #29: Add JUnit Testing
https://nus-cs2113-ay2021s1.github.io/website/schedule/week8/project.html#4-add-some-junit-tests

### PR DESCRIPTION
**Pull request by oasisbeatle**
JUnit Testing was added for methods in Resistor, Inductor, Capacitor and Voltage Source classes.

### CODE REVIEW

File: src/test/java/seedu/duke/component/CapacitorTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.junit.jupiter.api.Test;
+import seedu.duke.DukeException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CapacitorTest {
+
+    @Test
+    void getValueTest() {
```
Comment (hughjazzman) [inline src/test/java/seedu/duke/component/CapacitorTest.java]: Consider updating the method name to be more specific? Like https://se-education.org/guides/conventions/java/basic.html#naming
> `featureUnderTest_testScenario_expectedBehavior()`

Also can consider adding edge cases? Like 0 or negative values to make the tests more robust?

File: src/test/java/seedu/duke/component/CapacitorTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package seedu.duke.component;
+
+import org.junit.jupiter.api.Test;
+import seedu.duke.DukeException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CapacitorTest {
```
Comment (hughjazzman) [inline src/test/java/seedu/duke/component/CapacitorTest.java:8]: Perhaps the `addSeries()` and `addParallel()` methods could be tested as well? They are specific to the `LoadComponent` subclasses.

**Review by hughjazzman — CHANGES_REQUESTED** [review]
Just the comments to address, otherwise looks fine.

File: src/test/java/seedu/duke/component/CapacitorTest.java (hunk 2 above)
Comment (oasisbeatle) [reply src/test/java/seedu/duke/component/CapacitorTest.java:8]: Should I remove the previously created LoadComponentTest class and split it into the respective component test subclasses?

Comment (hughjazzman) [reply src/test/java/seedu/duke/component/CapacitorTest.java:8]: Oh that's a good point. No need to put them here then, you can just add test cases to `LoadComponentTest`?

Comment (oasisbeatle) [reply src/test/java/seedu/duke/component/CapacitorTest.java:8]: Yes, that's a good idea

**Comment by oasisbeatle** [comment]
Please let me know how if the way I've done LoadComponentTest so far makes sense, if it does I will extend it to cover all components

File: src/main/java/seedu/duke/component/Component.java [hunk 3]
```diff
@@ -24,7 +26,9 @@ public double getValue() {
      * @param value Value of Component.
      */
     public void setValue(double value) {
-        this.value = value;
+        if (value > 0) {
+            this.value = value;
+        }
```
Comment (hughjazzman) [inline src/main/java/seedu/duke/component/Component.java:31]: Could `else` `throw new DukeException` then?

File: src/test/java/seedu/duke/component/LoadComponentTest.java [hunk 4]
```diff
@@ -3,19 +3,32 @@
 import org.junit.jupiter.api.Test;
 import seedu.duke.DukeException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class LoadComponentTest {
```
Comment (hughjazzman) [inline src/test/java/seedu/duke/component/LoadComponentTest.java:9]: Yep looks good!

File: src/main/java/seedu/duke/component/Component.java (hunk 3 above)
Comment (oasisbeatle) [reply src/main/java/seedu/duke/component/Component.java:31]: Noted, thanks!

**Review by hughjazzman — APPROVED** [review]
LGTM
